### PR TITLE
[opt](iceberg)Add a new appearance to display the pushDown `count`

### PR DESCRIPTION
**Pull request by wuwenchi**
## Proposed changes

1. When the count can be pushed down, the specific data will be displayed later.
```

|      pushdown agg=COUNT (1)     |
```

2. Add a session variable `enable_count_push_down_for_external_table ` to control whether a pushdown count operation is required. Default is `true`;
```
mysql> show variables like 'enable_count_push_down_for_external_table';
+-------------------------------------------+-------+---------------+---------+
| Variable_name                             | Value | Default_Value | Changed |
+-------------------------------------------+-------+---------------+---------+
| enable_count_push_down_for_external_table | false | true          | 1       |
+-------------------------------------------+-------+---------------+---------+
1 row in set (0.02 sec)
```

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -1800,6 +1802,10 @@ public void setEnableLeftZigZag(boolean enableLeftZigZag) {
             description = {"强制使用jni方式读取外表", "Force the use of jni mode to read external table"})
     private boolean forceJniScanner = false;
 
+    @VariableMgr.VarAttr(name = OPTIMIZE_PUSH_DOWN_COUNT,
+            description = {"对`COUNT`使用下推进行优化", "Optimize `COUNT` using pushdown"})
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1806]:
```suggestion
            description = {"对外表启用 count(*) 下推优化", "enable count(*) pushdown optimization for external table"})
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -1800,6 +1802,10 @@ public void setEnableLeftZigZag(boolean enableLeftZigZag) {
...
+    private boolean optimizePushDownCount = true;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1807]:
```suggestion
    private boolean enableCountPushDownForExternalTable = true;
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 3]
```diff
@@ -584,6 +584,8 @@ public class SessionVariable implements Serializable, Writable {
 
     public static final String FORCE_JNI_SCANNER = "force_jni_scanner";
 
+    public static final String OPTIMIZE_PUSH_DOWN_COUNT = "optimize_push_down_count";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:587]:
```suggestion
    public static final String ENABLE_COUNT_PUSH_DOWN_FOR_EXTERNAL_TABLE = "enable_count_push_down_for_external_table";
```

**Review by morningman — COMMENTED** [review]
Should be used for all external table

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run feut

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zddr — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**